openedx_rocketchat/utils.py

```diff
@@ -0,0 +1,205 @@
+import hashlib
+import logging
+import re
+
+from django.conf import settings
+from django.core.cache import cache
+
+from openedx_rocketchat.edxapp_wrapper.get_configuration_helpers import get_configuration_helpers
+from openedx_rocketchat.edxapp_wrapper.get_student_models import anonymous_id_for_user, get_user
+
+from rocketchat_API.rocketchat import RocketChat as ApiRocketChat
+from rocketchat_API.APIExceptions.RocketExceptions import RocketAuthenticationException, RocketConnectionException
+
+
+LOG = logging.getLogger(__name__)
+CACHE_TIMEOUT = get_configuration_helpers().get_value("ROCKET_CHAT_CACHE_TIMEOUT", 86400)
+ROCKET_CHAT_DATA = "rocket_chat_data"
+
+
+def get_rocket_chat_settings():
+    """
+    Returns a dict with rocket chat settings. This try to get form the general settings and xblock settings.
+    **Example**:
+
+        ROCKET_CHAT_SETTINGS = {
+            'private_url_service': 'http://localhost:3000',
+            'public_url_service': 'localhost:3000',
+            'admin_user': 'rocketchat_admin_user',
+            "admin_pass": 'pass1234',
+            "username":"staff",
+            "password":"edx",
+        }
+
+        XBLOCK_SETTINGS = {
+            'RocketChatXBlock': {
+                'private_url_service': 'http://localhost:3000',
+                'public_url_service': 'localhost:3000',
+                'admin_user': 'rocketchat_admin_user',
+                "admin_pass": 'pass1234',
+                "username":"staff",
+                "password":"edx",
+            }
+        }
+    """
+    try:
+        return settings.ROCKET_CHAT_SETTINGS
+    except AttributeError, settings_error:
+        LOG.warning('Get settings warning: %s', settings_error)
+        pass
+    try:
+        xblock_settings = settings.XBLOCK_SETTINGS
+        return xblock_settings.get('RocketChatXBlock', None)
```

Comment:
You can get the value from a settings using: getattr(settings, "XBLOCK_SETTINGS", default-value) to avoid the try-catch block.